### PR TITLE
Use MacOS-12

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
We are excited to review your PR.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [ ] You have included any necessary tests in the same PR.
#6314

### CODE REVIEW

File: build/vsts-ci.yml [hunk 1]
```diff
@@ -119,7 +119,7 @@ jobs:
     DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
     DOTNET_MULTILEVEL_LOOKUP: 0
   pool:
-    vmImage: macOS-10.15
+    vmImage: macOS-12
   steps:
   # Work around MacOS Homebrew image/environment bug: https://github.com/actions/virtual-environments/issues/2322#issuecomment-749211076
```
Comment (ericstj) [inline build/vsts-ci.yml:124]: I wonder if we still need these workarounds?

**Review by ericstj — APPROVED** [review]
It looks reasonable to me - so long as it passes validation

**Review by dakersnar — APPROVED** [review]

File: build/vsts-ci.yml (hunk 1 above)
Comment (LittleLittleCloud) [reply build/vsts-ci.yml]: Looks like it's resolved. I just remove the workaround and kick off a new build

**Review by tarekgh — APPROVED** [review]

**Pull request merged**